adapters/handlers/grpc/v1/parse_search_request.go

```diff
@@ -55,21 +55,21 @@ func searchParamsFromProto(req *pb.SearchRequest, getClass func(string) *models.
 
 	out.Tenant = req.Tenant
 
-	targetVectors, targetCombination, err := extractTargetVectors(req, class)
+	targetVectors, targetCombination, vectorSearch, err := extractTargetVectors(req, class)
```

Comment:
maybe I am missing something, but why is this called vectorSearch? We are still doing a vectorsearch if there is no target vector given (as long as the collection has no named vectors or only one)

Reply:
We need to be able to discern between the two cases where the user is performing a vector search without specifying any target vectors and not performing a vector search at all. Since `targetVectors` is now `[]string` this means that "no target vectors" and "none target vectors" are identical, due to Go's logic. Previously, we used `*[]string` to discern these cases with `nil` corresponding to no vector search but since it's now `[]string` we need some other way to identify if no vector search is being performed. Luckily, you have the `vectorSearch` bool defined already for this exact purpose so I've added to where it was needed previously

Reply:
all good, I forgot how vectorSearch was defined and assume you would jsut return if target vectors are present or not